src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -91,10 +91,10 @@ private void OnTemplateChanged(DependencyPropertyChangedEventArgs e)
 		internal void SetUpdateControlTemplate(bool forceUpdate = false)
 		{
 			if (
-				!FeatureConfiguration.Control.UseLegacyLazyApplyTemplate
-				|| forceUpdate
-				|| this.HasParent()
-				|| CanCreateTemplateWithoutParent
+				(!FeatureConfiguration.Control.UseLegacyLazyApplyTemplate
+					|| forceUpdate
+					|| CanCreateTemplateWithoutParent)
+				&& this.HasParent() // Instead we should check if we are in a the visual tree
```

Comment:
Add a reference to #119

Reply:
I also linked the source bug of this PR into your epic since the bug cannot be resolved properly until the epic got completed.